src/dateparser/dateparser.js

```diff
@@ -82,6 +108,7 @@ angular.module('ui.bootstrap.dateparser', [])
     }
 
     format = $locale.DATETIME_FORMATS[format] || format;
+    format = format.replace(SPECIAL_CHARACTERS_REGEXP, '\\$&');
```

Comment:
Can you explain the less weird characters part?  Just moving this inside the `createParser` function isn't enough, there seems to be some specificity required as to when the parsing happens since the regex gets some additional characters through the iteration over `formatCodeToRegex`.  The loop adds an open and closed set of parentheses to the regex.  This means that it would be too late to modify the format string at that stage.